docs/testing.md

```diff
@@ -0,0 +1,179 @@
+    <Button
+      label="Add the item to list"
+      onPress={this.addNewItemToShoopingList}
+    />;
+  }
+
+  addNewItemToShoppingList = (someGroceryItem: string) => {
+    // ...
+  };
+}
+```
+
+When testing user interactions, test the component from the user perspective: you could write your tests to call `addNewItemToShoppingList` directly, and while such test works, it’s very close to the implementation details of your component.
+
+To counter for that, component testing libraries such as [`react-native-testing-library`](https://github.com/callstack/react-native-testing-library), offer `fireEvent` apis that simulate a user interacting with the component. There are apis that allow to simulate entering text into textinput, tapping buttons and more. An example of how you may fire text change event using `react-native-testing-library`:
+
+`fireEvent.changeText(getByTestId(‘groceryTextInput'), ‘banana’);`
+
+The first parameter is a query function that returns a `ReactTestInstance` and the second parameter is the new text that user entered.
+
+This way, we're not testing what some function is doing - we're testing what happens when a user changes the text in the `TextInput`!
+
+Now that we know how to fire an event, let's take a look at verifying that it caused the expected change in your component: for example, that a certain text is rendered in a `GroceryShoppingList`. Aforementioned component testing libraries expose `query` apis for this purpose. An example may look like this:
+
+```js
+import {render} from 'react-native-testing-library';
+
+const {getByText} = render(<GroceryShoppingList />); // render shopping list
+const bananaElement = getByText('banana');
+expect(bananaElement).toBeDefined(); // expect 'banana' to be on the list
+```
+
+### Testing rendered output
+
+[Snapshot testing](https://jestjs.io/docs/en/snapshot-testing) lets you test how—and if—a component is rendering. A snapshot is a textual representation of your component’s output that gets committed in your repo. When using snapshot testing, you first implement your component and then run the snapshot test, which saves the snapshot to a file. Any future changes to the component will result in a change in the snapshot that needs to be checked as part of your code review.
+
+In theory, you can use snapshots to test anything that is serializable, but do not overuse them! Snapshot is created _after_ you’re done working on the component, and at that point the snapshot is considered to be correct (even in the case when the rendered output is actually wrong). You need to make sure that the snapshot contains what you expect (which may not be trivial because snapshots can be fairly large). Snapshots do not ensure that your component render logic is correct, they are merely good at guarding against unexpected changes and for checking that the components in the React tree under test receive the expected props (styles and etc.).
```

Comment:
Typo at "Snapshot is created" Does it need a "The"?

Reply:
Sorry, I'm not sure what "The" you're referring to, but I have checked " Allow edits from maintainers." so maybe you can edit it right here in the PR.

Reply:
note that I expanded the snapshot testing paragraph in https://github.com/facebook/react-native-website/pull/1469/commits/127218a09cec6f58bcd4039825e567982eae64f0

Reply:
@vonovak We are using hooks and function components in the docs exclusively going forward. If you can do this, I can merge!

Reply:
snapshot part was rewritten with help of thymikee and class component is gone (example uses hooks) so resolving.